src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,117 @@
+import classNames from 'classnames';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
+type Props = {
+  todos: Todo[];
+  userId: number;
+  value: string;
+  sortBy: string;
+  randomSort: boolean;
+  selectUser: (x: number) => void;
+  changeValue: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  changeSortBy: (e: React.ChangeEvent<HTMLSelectElement>) => void;
+  changeRandomSort: () => void;
+};
 
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
+export const TodoList: React.FC<Props> = React.memo(
+  ({
+    todos,
+    userId,
+    value,
+    sortBy,
+    randomSort,
+    selectUser,
+    changeValue,
+    changeSortBy,
+    changeRandomSort,
+  }) => {
+    return (
+      <div className="TodoList">
+        <h2>Todos:</h2>
 
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
+        <div className="sorting-form TodoList__sorting-form">
+          <input
+            className="sorting-form__input"
+            type="text"
+            placeholder="Search todo"
+            value={value}
+            onChange={changeValue}
+          />
 
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
+          <div className="sorting-form__footer">
+            <select
+              className="button sorting-form__select"
+              value={sortBy}
+              onChange={changeSortBy}
+            >
+              <option value="">Select sorting option</option>
+              <option value="all">All todos</option>
+              <option value="completed">Completed todos</option>
+              <option value="active">Active todos</option>
+            </select>
+            <button
+              className={classNames(
+                'button',
+                'sorting-form__button',
+                { 'sorting-form__button--active': randomSort },
+              )}
+              type="button"
+              onClick={changeRandomSort}
+            >
+              Randomize
+            </button>
+          </div>
 
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
+        </div>
+
+        <div className="TodoList__list-container">
+          {todos.length > 0 ? (
+            <ul className="TodoList__list">
```

Comment:
TodoList component should contain only the list itself and nothing more, move the form to App component

Reply:
Ok, don't prfoblem, but in tasks was something like this:
![image](https://user-images.githubusercontent.com/92580512/164449971-4f11d471-c247-45fa-8572-41cd97e7ec4a.png)